### PR TITLE
Update Euler2RGB to resolve "nan" output for cubic symmetry

### PR DESCRIPTION
**Pull request by frombs**
Fixed bug, added error checks, and updated gold files of affected tests.
Closes issue #13443.

### CODE REVIEW

File: modules/phase_field/src/utils/Euler2RGB.C
Comment (dschwen) [inline modules/phase_field/src/utils/Euler2RGB.C]: Do we want this to fail hard? It seems to me that for an output capability we should soft fail (color the output in a specific way for failed matches).

File: modules/phase_field/src/utils/Euler2RGB.C
Comment (dschwen) [inline modules/phase_field/src/utils/Euler2RGB.C]: Again, do we need a hard fail here? (also typo `componenet`)

Comment (dschwen) [reply modules/phase_field/src/utils/Euler2RGB.C]: `red` can never be negative! For the other two I'd have to look closer at all those acos arguments, but this may be a case for an assertion instead.

File: modules/phase_field/src/utils/Euler2RGB.C
Comment (dschwen) [inline modules/phase_field/src/utils/Euler2RGB.C]: that comment does not make sense...

File: modules/phase_field/src/utils/Euler2RGB.C
Comment (frombs) [reply modules/phase_field/src/utils/Euler2RGB.C]: That is a good question. In an indirect way, the current code already soft fails by taking the solution from the last given symmetry operator and continuing along with incorrect results.  This is why the negative values made their way into the square root calculation. If the code is written correctly, we should never get a hard fail since a given set of valid Euler angles will always map into the standard stereographic triangle for a given crystal system. If they don't, I would rather know it so it can be fixed.  If you prefer to not have the hard fail, we could add a warning along with a specific output color.  Let me know what direction you want to go.

File: modules/phase_field/src/utils/Euler2RGB.C
Comment (frombs) [reply modules/phase_field/src/utils/Euler2RGB.C]: I guess this is kind of redundant.  If the Euler angles map into the SST correctly, these will never be negative. I can remove it or change is to a warning if you prefer.

File: modules/phase_field/src/utils/Euler2RGB.C
Comment (frombs) [reply modules/phase_field/src/utils/Euler2RGB.C]: I need to reword that line.  Let me know how you want to handle the other two issues and I will work on the changes.

**Comment by frombs** [comment]
@dschwen, I responded to your comments.  Please review and let me know how you want to proceed. Thanks!

File: modules/phase_field/src/utils/Euler2RGB.C
Comment (dschwen) [reply modules/phase_field/src/utils/Euler2RGB.C]: change to a `mooseAssert`

File: modules/phase_field/src/utils/Euler2RGB.C
Comment (dschwen) [reply modules/phase_field/src/utils/Euler2RGB.C]: I leave that up to you. Things that "should never happen" are usually covered by an assertion

**Comment by frombs** [comment]
@dschwen, I added the assertions and the checks have passed so this is ready for review.

File: modules/phase_field/src/utils/Euler2RGB.C
Comment (dschwen) [inline modules/phase_field/src/utils/Euler2RGB.C]: `!(index == nsym)` ? That's a peculiar way to write `index != nsym`, but it's fine

**Review by dschwen — DISMISSED** [review]
Looks ok now

File: modules/phase_field/src/utils/Euler2RGB.C
Comment (frombs) [reply modules/phase_field/src/utils/Euler2RGB.C]: Yep, I was a little tired when I did that.  I will fix it and push it to the branch.

**Comment by frombs** [comment]
@dschwen, I didn't realize you would need to re-approve this when I made the change. Sorry to cause you more work.

**Comment by frombs** [comment]
@dschwen, can we finalize this PR?

**Comment by dschwen** [comment]
Ugh, lost track of this. Could you please rebase?

**Comment by dschwen** [comment]
In #13778 @fdkong updated these inputs and gold files. You'll have to drop your changes, rebase onto a new devel and redo your runs to create the new gold files.

**Comment by frombs** [comment]
@dschwen, the rebase is complete and the checks have passed.

**Comment by frombs** [comment]
@dschwen, this PR needs your approval.

**Review by dschwen — APPROVED** [review]

**Pull request merged**
